analyzers/src/SonarAnalyzer.Common/Rules/Hotspots/CreatingHashAlgorithmsBase.cs

```diff
@@ -28,23 +28,48 @@ public abstract class CreatingHashAlgorithmsBase<TSyntaxKind> : SonarDiagnosticA
         protected const string DiagnosticId = "S4790";
         protected const string MessageFormat = "Make sure this weak hash algorithm is not used in a sensitive context here.";
 
-        protected ObjectCreationTracker<TSyntaxKind> ObjectCreationTracker { get; set; }
+        private const string CreateMethodName = "Create";
+
+        private readonly KnownType[] algorithmTypes =
+        {
+            KnownType.System_Security_Cryptography_DSA,
+            KnownType.System_Security_Cryptography_HMACMD5,
+            KnownType.System_Security_Cryptography_HMACRIPEMD160,
+            KnownType.System_Security_Cryptography_HMACSHA1,
+            KnownType.System_Security_Cryptography_MD5,
+            KnownType.System_Security_Cryptography_RIPEMD160,
+            KnownType.System_Security_Cryptography_SHA1
+        };
 
-        protected InvocationTracker<TSyntaxKind> InvocationTracker { get; set;  }
+        private readonly string[] unsafeAlgorithms = { "SHA1", "MD5", "DSA", "HMACMD5", "HMACRIPEMD160", "RIPEMD160", "RIPEMD160Managed" };
```

Comment:
Value `System.Security.Cryptography.SHA1` is also valid algorithm name according to [docs](https://docs.microsoft.com/en-us/dotnet/api/system.security.cryptography.hashalgorithm.create?view=netframework-4.8#System_Security_Cryptography_HashAlgorithm_Create_System_String_). All forms would be scattered around. We should support them or document FNs.

Reply:
The same concept applies to all:
https://docs.microsoft.com/en-us/dotnet/api/system.security.cryptography.hmac.create?view=netframework-4.8

I haven't found a list in docs for `RIPEMD160` and `RIPEMD160Managed` as the [docs](https://docs.microsoft.com/en-us/dotnet/api/system.security.cryptography.ripemd160.create?view=netframework-4.8#System_Security_Cryptography_RIPEMD160_Create_System_String_) are missing the list of valid values. But quick test shows, that:
`"System.Security.Cryptography.RIPEMD160"` creates instance of `RIPEMD160Managed` type, so we should support it.
`"RIPEMD160Managed"` is not a valid value for the argument, it returns `null`. So we should remove it.

Reply:
I'm sorry I missed that the full list of valid values is scattered around :/

Reply:
> I haven't found a list in docs for RIPEMD160 and RIPEMD160Managed as the docs are missing the list of valid values. But quick test shows, that:
"System.Security.Cryptography.RIPEMD160" creates instance of RIPEMD160Managed type, so we should support it.
"RIPEMD160Managed" is not a valid value for the argument, it returns null. So we should remove it.

They were present from the old implementation. I wonder now if it was a mistake or they were really needed.

Reply:
I've removed **RIPEMD160Managed** and added test cases with namespaces for the rest of the values